### PR TITLE
`useMergeRefs` custom hook

### PR DESCRIPTION
**Pull request by dmanek**
Carrying over from https://github.com/ampproject/amphtml/pull/35835#discussion_r705732157

### CODE REVIEW

File: src/preact/utils.js [hunk 1]
```diff
@@ -16,15 +16,31 @@ export function useResourcesNotify() {
...
+  }
+}
+
+/**
+ * Combines refs to pass into `ref` prop.
+ * @param {any} refOne
+ * @param {any} refTwo
+ * @return {function()}
+ */
+export function useMergeRefs(refOne, refTwo) {
```
Comment (dmanek) [inline src/preact/utils.js:36]: Through https://github.com/theKashey/use-callback-ref#usemergerefs I came across react-hooks.org. This is inspired from their `useForkRef` implementation. Tradeoff is it only combines 2 refs, but for now it does what is required.

File: src/preact/utils.js [hunk 2]
```diff
@@ -16,15 +16,31 @@ export function useResourcesNotify() {
 }
 
 /**
- * Combines multiple refs to pass into `ref` prop.
- * @param {...any} refs
- * @return {function(!Element)}
+ * @param {any} ref
```
Comment (samouri) [inline src/preact/utils.js:19]: The `any` isn't valid closure 😢 . Should be `*`

File: src/preact/utils.js [hunk 3]
```diff
@@ -16,15 +16,31 @@ export function useResourcesNotify() {
...
+}
+
+/**
+ * Combines refs to pass into `ref` prop.
+ * @param {any} refOne
+ * @param {any} refTwo
+ * @return {function()}
+ */
+export function useMergeRefs(refOne, refTwo) {
+  return useMemo(() => {
```
Comment (samouri) [inline src/preact/utils.js:37]: Why would a ref be null?

File: src/preact/utils.js [hunk 4]
```diff
@@ -16,15 +16,31 @@ export function useResourcesNotify() {
...
+    if (refOne === null && refTwo === null) {
+      return null;
     }
-  };
+    return (element) => {
```
Comment (samouri) [inline src/preact/utils.js:41]: nit: Seems even easier to take an array

```js
export function useMergeRefs(refs) {
  return useCallback(
    (element) => refs.forEach((ref) => setRef(ref, element)),
    refs
  );
}
```

File: src/preact/utils.js (hunk 3 above)
Comment (samouri) [inline src/preact/utils.js:37]: I think `useCallback` is slightly better sugar for this situation

File: src/preact/utils.js (hunk 2 above)
Comment (dmanek) [reply src/preact/utils.js]: Updated type to `{current: ?}|function()`

File: src/preact/utils.js (hunk 3 above)
Comment (dmanek) [reply src/preact/utils.js]: Removed.

Comment (dmanek) [reply src/preact/utils.js]: Done.

File: src/preact/utils.js (hunk 4 above)
Comment (dmanek) [reply src/preact/utils.js]: Updated, except used a native for loop since @jridgewell is not a fan of `forEach` 🙂

**Review by samouri — APPROVED** [review]

File: src/preact/utils.js (hunk 4 above)
Comment (samouri) [reply src/preact/utils.js]: you learn fast 😆

**Review by jridgewell — APPROVED** [review]

**Comment by ivannggnnavi** [comment]
J3jj

**Pull request merged**